### PR TITLE
Update DEPLOY.md

### PR DESCRIPTION
**Pull request by flotwig**
- Document kitchensink release process better
- Clarify pushing up the release commit

### CODE REVIEW

File: DEPLOY.md [hunk 1]
```diff
@@ -144,15 +145,17 @@ Once all test projects are reliably working with new changes, publishing can pro
     ```
     npm run binary-release -- --version 3.4.0 --commit`
     ```
-9. Tag the current commit with `v3.4.0` and push that tag up.
-10. If needed, push out the updated changes to the docs manifest to `on.cypress.io`.
-11. If needed, push out an updated kitchen sink.
-12. Close the release in [ZenHub](https://app.zenhub.com/workspaces/test-runner-5c3ea3baeb1e75374f7b0708/reports/release).
-13. Bump `version` in `package.json` from `develop` branch and then merge into `master`.
-14. Using [cypress-io/release-automations][release-automations]:
+9. If needed, push out the updated changes to the docs manifest to `on.cypress.io`.
```
Comment (jennifer-shehane) [inline DEPLOY.md]: Nice to mention this, I always forget this.

File: DEPLOY.md [hunk 2]
```diff
@@ -144,15 +145,17 @@ Once all test projects are reliably working with new changes, publishing can pro
...
+10. If needed, push out an updated [`cypress-example-kitchensink`][cypress-example-kitchensink] on `example.cypress.io`
```
Comment (jennifer-shehane) [inline DEPLOY.md]: I'm not sure how fully you want to document this here, but how would you push this out? Is there some script or something that should be run?

File: DEPLOY.md [hunk 3]
```diff
@@ -104,6 +104,7 @@ Once all test projects are reliably working with new changes, publishing can pro
 
 ### Steps to Publish a New Version
 
+0. Make sure that if there is a new [`cypress-example-kitchensink`][cypress-example-kitchensink] version, the corresponding dependency in `packages/example` has been updated to that new version.
```
Comment (jennifer-shehane) [inline DEPLOY.md]:
```suggestion
0. Make sure that if there is a new [`cypress-example-kitchensink`][https://github.com/cypress-io/cypress-example-kitchensink/releases] version, the corresponding dependency in `packages/example` has been updated to that new version.
```

File: DEPLOY.md (hunk 2 above)
Comment (flotwig) [reply DEPLOY.md]: I honestly don't know, I think Brian has a script or something, @brian-mann?

Comment (jennifer-shehane) [inline DEPLOY.md]:
```suggestion
10. If needed, deploy the updated [`cypress-example-kitchensink`][cypress-example-kitchensink] to `example.cypress.io` by following [these instructions under "Deployment"](./packages/example/README.md).
```

**Pull request merged**